code/SecureFileController.php

```diff
@@ -205,5 +205,35 @@ public function canDownloadFile(File $file = null) {
 		return false;
 	}
 
+    /**
+     * Case-sensitive version of File::find(). This is used because case-sensitive file systems can have files at both
+     * upload.jpg and upload.JPG. Without this change, we always return the first of those files no matter what. SS
+     * *shouldn't* allow two files to be uploaded with the same (case insensitive) filename, but it does.
+     *
+     * @param string $filename
+     * @return File|null
+     */
+    public function find($filename) {
+        // Get the base file if $filename points to a resampled file
+        $filename = Image::strip_resampled_prefix($filename);
```

Comment:
this method doesn't appear to be available in 3.1 and 3.2 which means that our test suite is failing...

I'm not against bumping our minimum requirement up to 3.3, but not everyone may agree

Reply:
We should bump this and allow users of 3.1 / 3.2 to continue to use older secureassets module.

Reply:
happy with that, @madmatt can you change the composer.json so we only support ^3.3?